Content.Shared/Item/ItemToggle/SharedItemToggleSystem.cs

```diff
@@ -0,0 +1,313 @@
+        if (!_light.TryGetLight(uid, out var light))
+            return;
+
+        _light.SetEnabled(uid, itemToggle.Activated, light);
+    }
+
+    /// <summary>
+    /// Used to update weapon component aspects, like hit sounds, damage values and hidden status while activated.
+    /// </summary>
+    private void UpdateWeaponComponent(EntityUid uid, ItemToggleMeleeWeaponComponent itemToggleMelee)
+    {
+        if (!TryComp(uid, out MeleeWeaponComponent? meleeWeapon))
+            return;
+
+        //Sets the damage values to the item's default if none is stated.
+        itemToggleMelee.ActivatedDamage ??= meleeWeapon.Damage;
+        itemToggleMelee.DeactivatedDamage ??= meleeWeapon.Damage;
+        //Sets the no damage sound to the item's default if none is stated.
+        itemToggleMelee.ActivatedSoundOnHitNoDamage ??= meleeWeapon.NoDamageSound;
+        itemToggleMelee.DeactivatedSoundOnHitNoDamage ??= meleeWeapon.NoDamageSound;
+        //Sets the swing sound to the item's default if none is stated.
+        itemToggleMelee.ActivatedSoundOnSwing ??= meleeWeapon.SwingSound;
+        itemToggleMelee.DeactivatedSoundOnSwing ??= meleeWeapon.SwingSound;
+
+
+        if (IsActivated(uid))
+        {
+            if (itemToggleMelee.ActivatedDamage != null)
+                meleeWeapon.Damage = itemToggleMelee.ActivatedDamage;
+
+            meleeWeapon.HitSound = itemToggleMelee.ActivatedSoundOnHit;
+
+            if (itemToggleMelee.ActivatedSoundOnHitNoDamage != null)
+                meleeWeapon.NoDamageSound = itemToggleMelee.ActivatedSoundOnHitNoDamage;
+
+            if (itemToggleMelee.ActivatedSoundOnSwing != null)
+                meleeWeapon.SwingSound = itemToggleMelee.ActivatedSoundOnSwing;
+
+            if (itemToggleMelee.DeactivatedSecret)
+                meleeWeapon.Hidden = false;
+        }
+        else
+        {
+            meleeWeapon.Damage = itemToggleMelee.DeactivatedDamage;
+
+            meleeWeapon.HitSound = itemToggleMelee.DeactivatedSoundOnHit;
+
+            if (itemToggleMelee.DeactivatedSoundOnHitNoDamage != null)
+                meleeWeapon.NoDamageSound = itemToggleMelee.DeactivatedSoundOnHitNoDamage;
+
+            if (itemToggleMelee.DeactivatedSoundOnSwing != null)
+                meleeWeapon.SwingSound = itemToggleMelee.DeactivatedSoundOnSwing;
+
+            if (itemToggleMelee.DeactivatedSecret)
+                meleeWeapon.Hidden = true;
+        }
+
+        Dirty(uid, meleeWeapon);
+    }
+
+    /// <summary>
+    /// Used to update item component aspects, like size values for items that expand when activated (heh).
+    /// </summary>
+    private void UpdateItemComponent(EntityUid uid, ItemToggleSizeComponent itemToggleSize)
+    {
+        if (!TryComp(uid, out ItemComponent? item))
+            return;
+
+        //Sets the deactivated size to the default if none is stated.
+        itemToggleSize.ActivatedSize ??= item.Size;
+        itemToggleSize.DeactivatedSize ??= item.Size;
+
```

Comment:
Setting the fields randomly here is a code smell, you should just use a local variable and null coalesce it instead of overwriting the null values on the component.

e.g.

var activateSize = itemToggleSize.ActivatedSize ?? item.Size;

then use that for setSize.

Also you're getting both fields even though you only use 1 at a time below under IsActivated.

Reply:
I've kinda rethought that and rewrote it. It's still not perfect - the point is to set the values to the item's size (the default from the yml) before the item's size gets replaced by the activated size (and similar for the others too). But now I only do that if there's an activated value to begin with. If there isn't, nothing happens.